src/lib/operators/aggregate_hash.hpp

```diff
@@ -77,7 +78,12 @@ struct AggregateResult {
 
   AccumulatorType accumulator{};
   size_t aggregate_count = 0;
-  RowID row_id{INVALID_CHUNK_ID, INVALID_CHUNK_OFFSET};
+
+  // A NULL_ROW_ID means that the aggregate result is not (yet) valid and should be skipped when materializing the
+  // results. There is no ambiguity with actual NULLs, because this field always points to a row from the input
```

Comment:
The member field of the struct. 

> As described above, this stores a pointer into the input data that is used to later restore the GROUP BY values.
A NULL_ROW_ID means that the aggregate result is not (yet) valid and should be skipped when materializing the
results. There is no ambiguity with actual NULLs because the aggregate operator is not NULL-producing. As such,
we know that each valid GROUP BY-group has at least one valid input RowID. As the input may be a ReferenceSegment, a valid RowID may *point to* a row that is NULL.

Better?

Reply:
Jep.